### PR TITLE
fix page title and metadata merge order

### PR DESCRIPTION
**Pull request by OsmondJiang**
fix: https://dev.azure.com/ceapex/Engineering/_workitems/edit/105452

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4891)

### CODE REVIEW

File: src/docfx/lib/json/JsonUtility.cs [hunk 1]
```diff
@@ -191,6 +191,17 @@ public static (List<Error> errors, JToken value) Parse(string json, string file)
...
+            if (overwrites == null)
+                return;
+
+            foreach (var overwrite in overwrites)
+            {
+                Merge(container, overwrite);
+            }
+        }
+
         public static void Merge(JObject container, JObject overwrite)
```
Comment (live1206) [inline src/docfx/lib/json/JsonUtility.cs]: should this method be `private` and only expose `Merge(JObject container, params JObject[] overwrites)`?

File: src/docfx/lib/json/JsonUtility.cs [hunk 2]
```diff
@@ -191,25 +190,14 @@ public static (List<Error> errors, JToken value) Parse(string json, string file)
             }
         }
 
-        public static void Merge(JObject container, JObject overwrite)
+        public static void Merge(JObject container, params JObject[] overwrites)
```
Comment (yufeih) [inline src/docfx/lib/json/JsonUtility.cs]: keep the original non-allocating Merge method

File: src/docfx/build/page/BuildPage.cs [hunk 3]
```diff
@@ -198,12 +196,13 @@ private static (List<Error> errors, JObject model)
                 errors.Add(Errors.HeadingNotFound(file));
             }
 
+            var (_, inputMetadata) = context.MetadataProvider.GetMetadata(file);
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: accumulate errors is

Comment (OsmondJiang) [reply src/docfx/build/page/BuildPage.cs]: it's calculated in build page, here we don't need do it.

**Comment by OsmondJiang** [comment]
image
Integration test failed caused by publishing, @fenxu would you please help to take a look?

File: src/docfx/build/page/BuildPage.cs (hunk 3 above)
Comment (yufeih) [reply src/docfx/build/page/BuildPage.cs]: I know, but it is a good practice to not relying on behaviors elsewhere.

**Comment by OsmondJiang** [comment]
The diffs are expected, same with PROD page:

image

**Pull request merged**